homework5/save_original_info.py

```diff
@@ -0,0 +1,61 @@
+"""
+Написать декоратор который позволит сохранять информацию из
+исходной функции (__name__ and __doc__), а так же сохранит саму
+исходную функцию в атрибуте __original_func
+
+print_result изменять нельзя, за исключением добавления вашего
+декоратора на строку отведенную под него - замените комментарий
+
+До применения вашего декоратор будет вызываться AttributeError при custom_sum.__original_func
+Это корректное поведение
+После применения там должна быть исходная функция
+
+Ожидаемый результат:
+print(custom_sum.__doc__)  # 'This function can sum any objects which have __add___'
+print(custom_sum.__name__)  # 'custom_sum'
+print(custom_sum.__original_func)  # <function custom_sum at <some_id>>
+"""
+
+import functools
+
+
+def save__name__doc__func(original_func):
+    def decorator(func):
+        def inner(*args, **kwargs):
+            return func(*args, **kwargs)
+
+        inner.__name__ = original_func.__name__
+        inner.__doc__ = original_func.__doc__
+        inner.__original_func = original_func
+        return inner
+
+    return decorator
+
+
+def print_result(func):
+    @save__name__doc__func(func)
```

Comment:
You can use `functools.wraps` instead. 
https://docs.python.org/3/library/functools.html#functools.wraps